test/network_analysis/conservation_laws.jl

```diff
@@ -333,14 +378,12 @@ let
         @test_nowarn XProblem(rn, u0, tspan, ps; remove_conserved_warn = false)
         @test_nowarn XProblem(rn, u0, tspan, ps; remove_conserved = true, remove_conserved_warn = false)
     end
-    # Conservation law removal currently not working for NonlinearSystems due to MTK depricating something. https://github.com/SciML/ModelingToolkit.jl/issues/3458, https://github.com/SciML/ModelingToolkit.jl/issues/3411
-    # Currently just checks whether a NonlinearProblem can be created.
-    @test_broken for XProblem in [NonlinearProblem, SteadyStateProblem]
+    for XProblem in [NonlinearProblem, SteadyStateProblem]
         XProblem(rn, u0, ps; remove_conserved = true)
-        # @test_nowarn XProblem(rn, u0, ps)
-        # @test_logs (:warn, r"You are creating a system or problem while eliminating conserved quantities. Please *") XProblem(rn, u0, ps; remove_conserved = true)
-        # @test_nowarn XProblem(rn, u0, ps; remove_conserved_warn = false)
-        # @test_nowarn XProblem(rn, u0, ps; remove_conserved = true, remove_conserved_warn = false)
+        @test_nowarn XProblem(rn, u0, ps)
+        @test_logs (:warn, r"You are creating a system or problem while eliminating conserved quantities. Please *") XProblem(rn, u0, ps; remove_conserved = true)
+        @test_nowarn XProblem(rn, u0, ps; remove_conserved_warn = false)
+        @test_nowarn XProblem(rn, u0, ps; remove_conserved = true, remove_conserved_warn = false)
     end
```

Comment:
These tests (and anything with `remove_conserved_warn`) is actually removed in https://github.com/SciML/Catalyst.jl/pull/1203